nifi-nar-bundles/nifi-spring-bundle/nifi-spring-processors/src/main/java/org/apache/nifi/spring/SpringContextProcessor.java

```diff
@@ -0,0 +1,382 @@
+        return relationships;
+    }
+
+    /**
+     *
+     */
+    @OnScheduled
+    public void initializeSpringContext(ProcessContext processContext) {
+        this.applicationContextConfigFileName = processContext.getProperty(CTX_CONFIG_NAME).getValue();
+        this.applicationContextLibPath = processContext.getProperty(CTX_LIB_PATH).getValue();
+
+        String stStr = processContext.getProperty(SEND_TIMEOUT).getValue();
+        this.sendTimeout = stStr == null ? 0 : FormatUtils.getTimeDuration(stStr, TimeUnit.MILLISECONDS);
+
+        String rtStr = processContext.getProperty(RECEIVE_TIMEOUT).getValue();
+        this.receiveTimeout = rtStr == null ? 0 : FormatUtils.getTimeDuration(rtStr, TimeUnit.MILLISECONDS);
+
+        try {
+            if (logger.isDebugEnabled()) {
+                logger.debug(
+                        "Initializing Spring Application Context defined in " + this.applicationContextConfigFileName);
+            }
+            this.exchanger = SpringContextFactory.createSpringContextDelegate(this.applicationContextLibPath,
+                    this.applicationContextConfigFileName);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed while initializing Spring Application Context", e);
+        }
+        if (logger.isInfoEnabled()) {
+            logger.info("Successfully initialized Spring Application Context defined in "
+                    + this.applicationContextConfigFileName);
+        }
+    }
+
+    /**
+     * Will close the 'exchanger' which in turn will close both Spring
+     * Application Context and the ClassLoader that loaded it allowing new
+     * instance of Spring Application Context to be created upon the next start
+     * (which may have an updated classpath and functionality) without
+     * restarting NiFi.
+     */
+    @OnStopped
+    public void closeSpringContext(ProcessContext processContext) {
+        if (this.exchanger != null) {
+            try {
+                if (logger.isDebugEnabled()) {
+                    logger.debug(
+                            "Closing Spring Application Context defined in " + this.applicationContextConfigFileName);
+                }
+                this.exchanger.close();
+                if (logger.isInfoEnabled()) {
+                    logger.info("Successfully closed Spring Application Context defined in "
+                            + this.applicationContextConfigFileName);
+                }
+            } catch (IOException e) {
+                getLogger().warn("Failed while closing Spring Application Context", e);
+            }
+        }
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession processSession) throws ProcessException {
+        FlowFile flowFileToProcess = processSession.get();
+        if (flowFileToProcess != null) {
+            this.sendToSpring(flowFileToProcess, context, processSession);
```

Comment:
@olegz feels like perhaps we should catch some sort of exception from here and route the flow file to a failure relationship.  If not then it means anytime a send to spring causes an exception we will rollback the session.  It is *possibly* better to instead have a failure relationship where we can penalize flow files and send them in the event they are things we send to spring and get a failure in doing so.